samples/interactive-tutorials/src/main/java/events/ImportUserEventsGcs.java

```diff
@@ -34,91 +34,95 @@
-        Operation operation = operationsClient.getOperation(operationName);
-
-        while (!operation.getDone()) {
-          // Keep polling the operation periodically until the import task is done.
-          int awaitDuration = 30000;
-          Thread.sleep(awaitDuration);
-          operation = operationsClient.getOperation(operationName);
-        }
-
-        if (operation.hasMetadata()) {
-          ImportMetadata metadata = operation.getMetadata().unpack(ImportMetadata.class);
-          System.out.printf(
-              "Number of successfully imported events: %s\n", metadata.getSuccessCount());
-          System.out.printf(
-              "Number of failures during the importing: %s\n", metadata.getFailureCount());
-        }
-
-        if (operation.hasResponse()) {
-          ImportUserEventsResponse response =
-              operation.getResponse().unpack(ImportUserEventsResponse.class);
-          System.out.printf("Operation result: %s%n", response);
-        }
-      } catch (InvalidArgumentException e) {
+    String gcsBucket = String.format("gs://%s", bucketName);
+    String gcsErrorsBucket = String.format("%s/error", gcsBucket);
+
+    GcsSource gcsSource =
+        GcsSource.newBuilder()
+            .addInputUris(String.format("%s/%s", gcsBucket, gcsUserEventsObject))
+            .build();
+
+    UserEventInputConfig inputConfig =
+        UserEventInputConfig.newBuilder().setGcsSource(gcsSource).build();
+
+    System.out.println("GRS source: " + gcsSource.getInputUrisList());
+
+    ImportErrorsConfig errorsConfig =
+        ImportErrorsConfig.newBuilder().setGcsPrefix(gcsErrorsBucket).build();
+
+    ImportUserEventsRequest importRequest =
+        ImportUserEventsRequest.newBuilder()
+            .setParent(defaultCatalog)
+            .setInputConfig(inputConfig)
+            .setErrorsConfig(errorsConfig)
+            .build();
+    System.out.printf("Import user events from google cloud source request: %s%n", importRequest);
+
+    // Initialize client that will be used to send requests. This client only
+    // needs to be created once, and can be reused for multiple requests. After
+    // completing all of your requests, call the "close" method on the client to
+    // safely clean up any remaining background resources.
+    try (UserEventServiceClient serviceClient = UserEventServiceClient.create()) {
+      String operationName = serviceClient.importUserEventsCallable().call(importRequest).getName();
+
+      System.out.println("The operation was started.");
+      System.out.printf("OperationName = %s%n", operationName);
+
+      OperationsClient operationsClient = serviceClient.getOperationsClient();
+      Operation operation = operationsClient.getOperation(operationName);
+
+      long assuredBreak = System.currentTimeMillis() + 60000; // 60 seconds delay
+
+      while (!operation.getDone() || System.currentTimeMillis() < assuredBreak) {
```

Comment:
Use java.time here like you did in the other samples

Reply:
Done